### ISSUE TITLE
Using CSS or JS Helpers. Cannot read property 'substring' of undefined.

### ISSUE THREAD

**Issue by ryan-sandy** [jaredly/hexo-renderer-handlebars#8]
If I try to use the CSS or JS helpers I get an error.

```
{{css "css/styles.css"}}
```

The file is located in `/themes/myTheme/source/css/styles.css`

What am I doing wrong?

**Comment by jaredly** [comment]
Do you have a full stack trace?

**Comment by mAAdhaTTah** [comment]
I've got a stack trace for this issue:

```
Unhandled rejection TypeError: path.substring is not a function
    at Object.cssHelper (/path/to/project/node_modules/hexo/lib/plugins/helper/css.js:15:16)
    at Object.wrapper [as css] (/path/to/project/node_modules/lodash/lodash.js:4941:19)
    at Object.eval (eval at createFunctionContext (/path/to/project/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:10:79)
    at main (/path/to/project/node_modules/handlebars/dist/cjs/handlebars/runtime.js:173:32)
    at ret (/path/to/project/node_modules/handlebars/dist/cjs/handlebars/runtime.js:176:12)
    at ret (/path/to/project/node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:525:21)
    at Hexo.handlebarsRenderer (/path/to/project/node_modules/hexo-renderer-handlebars/index.js:34:10)
    at Hexo.tryCatcher (/path/to/project/node_modules/bluebird/js/release/util.js:16:23)
    at Hexo.<anonymous> (/path/to/project/node_modules/bluebird/js/release/method.js:15:34)
    at /path/to/project/node_modules/hexo/lib/hexo/render.js:61:21
    at tryCatcher (/path/to/project/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/path/to/project/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/path/to/project/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromiseCtx (/path/to/project/node_modules/bluebird/js/release/promise.js:604:10)
    at Async._drainQueue (/path/to/project/node_modules/bluebird/js/release/async.js:143:12)
    at Async._drainQueues (/path/to/project/node_modules/bluebird/js/release/async.js:148:10)
    at Immediate.Async.drainQueues (/path/to/project/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:574:20)
    at tryOnImmediate (timers.js:554:5)
    at processImmediate [as _immediateCallback] (timers.js:533:5)
```

**Comment by mAAdhaTTah** [comment]
I dug into this a bit; the stack trace is weird because it doesn't appear to pass through the registered Handlebars helper before hitting the hexo helper. I commented out the registration and reran `hexo generate` and it doesn't appear to affect anything, so I'm not sure what's going on there.

The direct cause is the last argument passed is the Handlebars options object, which is neither a string nor an array, so it attempts to call `substring` on it and bombs out.

**Comment by klaronix** [comment]
Yep. Something changed in the way, helpers need to be registered (different context). Please check my open pull request (or the fork https://github.com/klaronix/hexo-renderer-handlebars). Fixed it for me.

**Comment by mAAdhaTTah** [comment]
@klaronix Using your fork resolved the issue for me, thanks!

**Comment by jaredly** [comment]
fixed by d702fe87ed6231dd7a2f46aa7abefffaefc56b27
